### PR TITLE
Add support for 64-bit RISC-V arch

### PR DESCRIPTION
**Pull request by rebeccasf**

### CODE REVIEW

**Comment by rebeccasf** [comment]
Hi @jonsneyers @mo271 @sboukortt @veluca93, I would like to request for review on the changes in this PR to enable RISCV arch. Thanks.

**Comment by mo271** [comment]
Thanks for your contribution!
`User Rebecca Chang Swee Fun <rebecca.chang@starfivetech.com> not found, please add yourself to the AUTHORS file
`

**Review by veluca93 — DISMISSED** [review]

**Comment by sboukortt** [comment]
I think the “authors” check is failing because now the merge commit itself is associated with a different e-mail address. Could you perhaps rebase your branch on `main` (instead of merging), which will get rid of the merge commit altogether?

File: lib/jxl/base/arch_macros.h [hunk 1]
```diff
@@ -28,6 +28,12 @@ namespace jxl {
 #define JXL_ARCH_ARM 0
 #endif
 
+#if defined(__riscv) && __riscv_xlen == 64
+#define JXL_ARCH_RISCV64 1
```
Comment (eustas) [inline lib/jxl/base/arch_macros.h:32]: Where this macro will be used?

Comment (rebeccasf) [reply lib/jxl/base/arch_macros.h:32]: I was working on Chromium build and got some errors when compiling libjxl, this changes will help Chromium compilation but we have yet to test the browser functionality with libjxl, we might need this in future enhancement for Chromium browser.

Comment (malaterre) [reply lib/jxl/base/arch_macros.h:32]: I am also interested in the build error, since libjxl compiles just fine on riscv64/Debian/sid. Thanks

Comment (malaterre) [reply lib/jxl/base/arch_macros.h:32]: Technically I see that `JXL_ARCH_PPC` is not used anywhere...

Comment (rebeccasf) [reply lib/jxl/base/arch_macros.h:32]: Thanks for you note, I have rebased the Chromium source code and retried. libjxl is compiling fine there. Thanks. I will close this PR.

**Pull request closed without merging**